### PR TITLE
test: Allow charge information and charge price updates without changes to existing data

### PR DESCRIPTION
**Pull request by HenrikSommer**
## Description

With this pull request, we ensure that:
- a Charge Information (D18) request with the exact same master data as stored is accepted and processed
- a Charge Price (D08) request with the same price values as stored is accepted and processed
without any changes to the stored data.

Note for the reviewer: It is accepted that ChargePeriod Id's are changed.

## References

* #1135

### CODE REVIEW

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/IntegrationTests/EndpointTests/ChargeCommandReceiverEndpointTests.cs [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+                    .WithChargeType(chargeType)
+                    .WithTransparentInvoicing(TransparentInvoicing.NonTransparent)
+                    .WithStartDateTime(startDateTime)
+                    .WithEndDateTime(null)
+                    .WithTaxIndicator(TaxIndicator.NoTax)
+                    .WithVatClassification(VatClassification.NoVat)
+                    .WithChargeName("Grid Access Provider test tariff")
+                    .WithDescription("Description...")
+                    .Build();
+                var priceCommand = commandBuilder
```
Comment (prtandrup) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/IntegrationTests/EndpointTests/ChargeCommandReceiverEndpointTests.cs:143]:
```suggestion
                var chargeCommand = commandBuilder
```

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/IntegrationTests/EndpointTests/ChargePriceCommandReceiverEndpointTests.cs [hunk 2]
```diff
@@ -83,6 +93,48 @@ public Task InitializeAsync()
                 actualOutboxMessage.Type.Should().Be(_operationsRejectedEventType);
             }
 
+            [Theory]
+            [InlineAutoMoqData]
+            public async Task RunAsync_WhenChargeIsNotChanged_ConfirmationEventIsRaised(
```
Comment (prtandrup) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/IntegrationTests/EndpointTests/ChargePriceCommandReceiverEndpointTests.cs:98]:
```suggestion
            public async Task RunAsync_WhenChargePriceIsNotChanged_ConfirmationEventIsRaised(
```

OR 
```suggestion
            public async Task RunAsync_WhenPriceSeriesIsNotChanged_ConfirmationEventIsRaised(
```

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/IntegrationTests/EndpointTests/ChargePriceCommandReceiverEndpointTests.cs [hunk 3]
```diff
@@ -83,6 +93,48 @@ public Task InitializeAsync()
...
+                // Assert
+                await using var actualChargeContext = Fixture.ChargesDatabaseManager.CreateDbContext();
+                var actualCharge = await actualChargeContext.Charges.FirstAsync(
+                    GetChargePredicate(chargeId, ownerId, chargeType));
+
+                using var assertionScope = new AssertionScope();
+                actualCharge.Should().BeEquivalentTo(expectedCharge);
+                var actualOutboxMessage = await actualChargeContext.OutboxMessages
+                    .SingleOrDefaultAsync(x => x.CorrelationId.Contains(correlationId));
+                actualOutboxMessage.Should().BeNull();
```
Comment (prtandrup) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/IntegrationTests/EndpointTests/ChargePriceCommandReceiverEndpointTests.cs:135]: Sneaky, It'll be easy to spot and refactor when new price flow actually does the confirmation part 👍

**Review by prtandrup — APPROVED** [review]
Approved with a few minor suggestions. And nice boy-scouting along the way 👍

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/IntegrationTests/EndpointTests/ChargePriceCommandReceiverEndpointTests.cs (hunk 3 above)
Comment (HenrikSommer) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/IntegrationTests/EndpointTests/ChargePriceCommandReceiverEndpointTests.cs:135]: Exactly :-)

**Pull request merged**